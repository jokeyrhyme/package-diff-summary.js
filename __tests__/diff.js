/* @flow */
'use strict';

const diff = require('../lib/diff.js');

test('deltaToMarkdown()', () => {
  const delta = {
    dependencies: [
      { name: 'added', version: '1.2.3', operation: 'new' },
      { name: 'dropped', version: '', operation: 'delete' },
      { name: 'rolledback', version: '~1.2.3', operation: 'edit' },
      { name: 'updated', version: '^2.3.4', operation: 'edit' },
      { name: 'execa', version: '2.3.4', operation: 'edit' },
      { name: 'package-diff-summary', version: '1.2.0', operation: 'edit' },
    ],
  };
  const oldPkg = {
    dependencies: {
      dropped: '1.2.3',
      rolledback: '~2.3.4',
      updated: '^1.2.3',
      execa: '1.2.3',
      'package-diff-summary': '1.1.0',
    },
  };
  return diff
    .deltaToMarkdown(delta, oldPkg)
    .then(result => expect(result).toMatchSnapshot());
});

test('diffPackages()', () => {
  const oldPkg = {
    dependencies: {
      dropped: '1.2.3',
      rolledback: '~2.3.4',
      updated: '^1.2.3',
    },
  };
  const pkg = {
    dependencies: {
      added: '1.2.3',
      rolledback: '~1.2.3',
      updated: '^2.3.4',
    },
  };
  const result = diff.diffPackages(oldPkg, pkg);
  expect(result).toMatchSnapshot();
});

test('versionToMarkdown("1.2.3", "CHANGELOG.md")', () => {
  const result = diff.versionToMarkdown('1.2.3', 'CHANGELOG.md');
  expect(result).toMatchSnapshot();
});

// http://semver.org/#spec-item-9
test('versionToMarkdown("1.2.3-alpha.1", "CHANGELOG.md")', () => {
  const result = diff.versionToMarkdown('1.2.3-alpha.1', 'CHANGELOG.md');
  expect(result).toMatchSnapshot();
});

// http://semver.org/#spec-item-9
test('versionToMarkdown("1.0.0-x.7.z.92", "CHANGELOG.md")', () => {
  const result = diff.versionToMarkdown('1.0.0-x.7.z.92', 'CHANGELOG.md');
  expect(result).toMatchSnapshot();
});
