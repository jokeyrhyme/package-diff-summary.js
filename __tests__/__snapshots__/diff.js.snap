// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deltaToMarkdown() 1`] = `
"

-   depend upon added 1.2.3

-   no longer depend upon dropped

-   rollback rolledback to 1.2.3 (from 2.3.4)

-   update [updated](https://www.npmjs.com/package/updated) to 2.3.4 (from [1.2.3](https://github.com/ahmadnassri/node-updated/releases/tag/v1.2.3))

-   update [execa](https://www.npmjs.com/package/execa) to 2.3.4 (from 1.2.3)

-   update [package-diff-summary](https://www.npmjs.com/package/package-diff-summary) to [1.2.0](https://github.com/jokeyrhyme/package-diff-summary.js/releases/tag/1.2.0) (from [1.1.0](https://github.com/jokeyrhyme/package-diff-summary.js/releases/tag/1.1.0))
"
`;

exports[`diffPackages() 1`] = `
Object {
  "dependencies": Array [
    Object {
      "cmd": "npm uninstall dropped",
      "name": "dropped",
      "operation": "delete",
      "version": "1.2.3",
    },
    Object {
      "cmd": "npm install \\"rolledback@~1.2.3\\" --spin=false --color=true --production --loglevel=http",
      "name": "rolledback",
      "operation": "edit",
      "version": "~1.2.3",
    },
    Object {
      "cmd": "npm install \\"updated@^2.3.4\\" --spin=false --color=true --production --loglevel=http",
      "name": "updated",
      "operation": "edit",
      "version": "^2.3.4",
    },
    Object {
      "cmd": "npm install \\"added@1.2.3\\" --spin=false --color=true --production --loglevel=http",
      "name": "added",
      "operation": "new",
      "version": "1.2.3",
    },
  ],
  "devDependencies": Array [],
  "optionalDependencies": Array [],
}
`;

exports[`versionToMarkdown("1.0.0-x.7.z.92", "CHANGELOG.md") 1`] = `"[1.0.0-x.7.z.92](CHANGELOG.md)"`;

exports[`versionToMarkdown("1.2.3", "CHANGELOG.md") 1`] = `"[1.2.3](CHANGELOG.md)"`;

exports[`versionToMarkdown("1.2.3-alpha.1", "CHANGELOG.md") 1`] = `"[1.2.3-alpha.1](CHANGELOG.md)"`;
